test/e2e/kubernetes/deployment/deployment.go

```diff
@@ -1,3 +1,4 @@
+//go:build e2e
```

Comment:
I don't think we want your change (maybe your IDE is set for < golang 1.17?)

Reply:
Right, it's the opposite of what I said.

@CecileRobertMichon @mboersma do we care about these changes leaking in? Or would we rather do them all in one go in a discrete PR? (I assume we want to prefer go 1.17+ idioms going forward)

Reply:
it's fine for them to leak in

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2130 should catch the rest